tests/foreman/api/test_http_proxy.py

```diff
@@ -125,3 +129,80 @@ def test_positive_end_to_end(setup_http_proxy, module_target_sat, module_manifes
     )
     assert len(yum_repo['output']) >= 1
     assert 'quay/busybox' in yum_repo['output']
+
+
+@pytest.mark.upgrade
+@pytest.mark.run_in_one_thread
+@pytest.mark.parametrize(
+    'setup_http_proxy',
+    [None, True, False],
+    indirect=True,
+    ids=['no_http_proxy', 'auth_http_proxy', 'unauth_http_proxy'],
+)
+@pytest.mark.tier3
+def test_positive_auto_attach(setup_http_proxy, module_target_sat, function_org, rhel8_contenthost):
+    """Attempt to auto attach a subscription to content host
+
+    :id: cce888b5-e023-4ee2-bffe-efa9260224ee
+
+    :customerscenario: true
+
+    :expectedresults: host successfully subscribed, subscription
+        repository enabled, and repository package installed.
+
+    :Assignee: jpathan
+
+    :BZ: 2046337
+
+    :parametrized: yes
+
+    :CaseLevel: System
+    """
+    with manifests.clone() as manifest:
+        upload_manifest(function_org.id, manifest.content)
+    lce = module_target_sat.api.LifecycleEnvironment(organization=function_org).create()
+    content_view = module_target_sat.api.ContentView(organization=function_org).create()
+    rh_repo_id = enable_sync_redhat_repo(constants.REPOS['rhel8_bos'], function_org.id)
+    rh_repo = module_target_sat.api.Repository(id=rh_repo_id).read()
+    assert rh_repo.content_counts['rpm'] >= 1
+    cv = module_target_sat.api.ContentView(id=content_view.id, repository=[rh_repo]).update(
+        ["repository"]
+    )
+    cv.publish()
+    cv = cv.read()
+    cv.version[-1].promote(data={'environment_ids': lce.id})
+    subscription = module_target_sat.api.Subscription(organization=function_org.id).search(
+        query={'search': f'name="{constants.DEFAULT_SUBSCRIPTION_NAME}"'}
+    )
+    assert len(subscription)
+    activation_key = module_target_sat.api.ActivationKey(
+        content_view=cv,
+        organization=function_org,
+        environment=lce,
+        auto_attach=False,
+    ).create()
+    activation_key.add_subscriptions(data={'subscription_id': subscription[0].id})
+    rhel8_contenthost.install_katello_ca(module_target_sat)
+    rhel8_contenthost.register_contenthost(
+        org=function_org.name, activation_key=activation_key.name
+    )
+    assert rhel8_contenthost.subscribed
+    # To Do: Use api
+    module_target_sat.cli.Host.subscription_register(
+        {
+            'organization-id': function_org.id,
+            'content-view-id': content_view.id,
```

Comment:
using `cv` and `content_view` seems a bit confusing to me, in the end, it is the same content view. Please pick one and use it in the whole test.